c/cardano_lock.c

```diff
@@ -399,12 +402,21 @@ int simulator_main() {
 #else
 int main(int argc, const char* argv[]) {
 #endif
+
+#ifdef CKB_C_STDLIB_PRINTF
+  printf("print is enable!\n");
+#endif  // CKB_C_STDLIB_PRINTF
   int err = CKB_SUCCESS;
-  uint8_t payment_pubkey[BLAKE2B_BLOCK_SIZE] = {0};
-  uint8_t stake_pubkey[BLAKE2B_BLOCK_SIZE] = {0};
-  err = get_args(payment_pubkey, stake_pubkey);
+  uint8_t header_type = 0;
+  uint8_t payment_pubkey[BLAKE2B_224_BLOCK_SIZE] = {0};
+  err = get_args(&header_type, payment_pubkey);
   CHECK(err == CKB_SUCCESS, err);
 
+  header_type = header_type >> 4;
+  CHECK((header_type == 0b0000 || header_type == 0b0010 ||
+         header_type == 0b0100 || header_type == 0b0110),
+        ERROR_UNSUPPORTED_ARGS);
+
```

Comment:
Check args size. It should be no less than 57. For 0b0110, it should be no less than 29.